components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/builtins.adoc

```diff
@@ -0,0 +1,152 @@
+= Cairo Builtins
+
+== Introduction
+
+Builtins are predefined optimized low-level execution units which are added to the Cairo CPU board to perform predefined computations which are expensive to perform in standard Cairo. Builtins enhance the functionality of the Cairo virtual machine, allowing you to perform tasks such as Pedersen hash, Poseidon hash, range-checks, ECDSA signature verifications, Keccak hash, Bitwise operations, and EC operations efficiently and at a cheaper gas cost.
```

Comment:
```suggestion
_Builtins_, in Cairo, are predefined optimized low-level execution units that the Cairo VM uses to perform predefined computations that are expensive to perform in standard Cairo. Builtins enhance the functionality of the Cairo VM, enabling you to perform tasks such as using the Poseidon hash, range-checks, or ECDSA signature verifications, efficiently and using less gas.
```
* _Allow_- means that you are not prevented from doing something. In contrast, _enable_ means that you can do something that you couldn't have done otherwise.
* _which_ vs _that_: In American English, which is the industry standard in technical documentation (as opposed to British English), _which_ sets off a part of a sentence that you remove entirely and the sentence would still make sense. A clause beginning with _that_ is an intrinsic part of the sentence.